src/App.js

```diff
@@ -13,15 +22,21 @@ function App() {
             <li>
               <Link to="/">Dashboard</Link>
             </li>
-            <li>
-              <Link to="/login/">Login</Link>
-            </li>
+            {user ? (
+              <li>
+                <button onClick={logout}>Log out</button>
+              </li>
+            ) : (
+              <li>
+                <Link to="/login/">Log in</Link>
+              </li>
+            )}
           </ul>
         </nav>
 
         <Route path="/" exact component={Dashboard} />
         <Route path="/accounts/new" component={CreateAccount} />
-        <Route path="/login/" component={Login} />
+        <Route path="/login/" render={() => <Login setUser={setUser} />} />
```

Comment:
Just a question: should we be passing `...props` to the components to give React Router access to the history api and such?

Reply:
makes sense, I didnt know it worked like that. I`ll include it